### PR TITLE
Default context is erase before backend exits in libusb_exit

### PR DESCRIPTION
**Pull request by liangyunwang**
This pull request address issue 554 by keeping default_context_lock till the end.

### CODE REVIEW

File: libusb/core.c [hunk 1]
```diff
@@ -2379,9 +2380,18 @@ void API_EXPORTED libusb_exit(struct libusb_context *ctx)
...
-	usbi_mutex_static_unlock(&default_context_lock);
+
+    /*
+     * Setting this flag without unlocking the default context, as
+     * we are actually destroying the default context.
+     * usbi_default_context is not set to NULL yet, as all activities
+     * would only stop after usbi_backend->exit() returns.
+     */
+    destroying_default_context = 1;
+	} else {
```
Comment (hjelmn) [inline libusb/core.c]: Indentation looks off here.

Comment (liangyunwang) [reply libusb/core.c]: oops! fixed in new commit.

**Comment by hjelmn** [comment]
Please squish into a single commit.

**Comment by liangyunwang** [comment]
Done. (hope I'm doing it right :p )

**Pull request closed without merging**